### PR TITLE
Adjust serialization

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/12647

### Changelog

* move serialization related files into cocos/serialization

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/serialization/ccon.ts [hunk 1]
```diff
@@ -230,6 +229,6 @@ export class BufferBuilder {
     }
 }
 
-legacyCC.internal.parseCCONJson = parseCCONJson;
-legacyCC.internal.decodeCCONBinary = decodeCCONBinary;
-legacyCC.internal.CCON = CCON;
+cclegacy.internal.parseCCONJson = parseCCONJson;
+cclegacy.internal.decodeCCONBinary = decodeCCONBinary;
```
Comment (dumganhar) [inline cocos/serialization/ccon.ts:233]: Any reason of renaming `legacyCC` to `cclegacy`?

Comment (minggo) [reply cocos/serialization/ccon.ts:233]: @PPpro said, it is reported as `cclegacy` to developers.

Comment (dumganhar) [inline cocos/serialization/ccon.ts:233]: Why rename `legacyCC` to `cclegacy`?

Comment (minggo) [reply cocos/serialization/ccon.ts:233]: In cc.d.ts, it is exported as cclegacy. So also change engine internal usage too.

**Review by dumganhar — APPROVED** [review]

File: cocos/core/data/object.ts [hunk 2]
```diff
@@ -343,7 +343,8 @@ class CCObject implements EditorExtendableObject {
      *       _destruct: function () {
      *           for (var key in this) {
      *               if (this.hasOwnProperty(key)) {
-     *                   switch (typeof this[key]) {
+     *                   switch (typeo
+     * f this[key]) {
```
Comment (PPpro) [inline cocos/core/data/object.ts]: why change this into two lines

Comment (minggo) [reply cocos/core/data/object.ts]: Good catch, i didn't modify it. May be the VSC modified it. I will fix it.

Comment (minggo) [reply cocos/core/data/object.ts]: fixed

**Pull request merged**